SSA/Projects/InstCombine/scripts/mlir-tool.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.dialects.llvm import LLVM
+from xdsl.utils.exceptions import ParseError
+from xdsl.context import MLContext
+from xdsl.dialects import get_all_dialects
+from xdsl.dialects.llvm import FuncOp
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.dialects.builtin import (
+    Builtin,
+    IndexType,
+    IntegerAttr,
+    IntegerType,
+    ModuleOp,
+    StringAttr,
+    i32,
+    i64,
+)
+import os
+import io
+from pathlib import Path
+from xdsl.printer import Printer
+# Initialize the MLIR context and register the LLVM dialect
+ctx = MLContext(allow_unregistered = True)
+ctx.load_dialect(LLVM)
+ctx.load_dialect(Builtin)
+
+directory = os.fsencode("../vcombined-mlir")
+
+
+allowed_names = {
+    "llvm.return",
+    "llvm.mul",
+    "llvm.add",
+    "llvm.sub",
+    "llvm.shl",
+    "llvm.and",
+    "llvm.or",
+    "llvm.xor",
+    "llvm.mlir.constant",
+    "llvm.lshr",
+    "llvm.ashr",
+    "llvm.urem",
+    "llvm.srem",
+    "llvm.add",
+    "llvm.mul",
+    "llvm.sub",
+    "llvm.sdiv"
+}
+allowed_unregistered = set() #{
+    # "llvm.icmp"
+# }
+def allowed(op):
+        return hasattr(op, "sym_name") or (op.name== "builtin.unregistered" and op.op_name.data in allowed_unregistered)  or (op.name in allowed_names)
+def show(block):
+        output = io.StringIO()
+        p = Printer(stream = output)
+        block.print(p)
+        contents = output.getvalue()
+        output.close()
+        return contents
+def showr(region):
+        output = io.StringIO()
+        p = Printer(stream = output)
+        p.print_region(region)
+        contents = output.getvalue()
+        output.close()
+        return contents
```

Comment:
I'd expect str(block) and str(region) to be equivalent, f"{block}" or f"{region}" otherwise

Reply:
In theory, yes. But I tried that and I got


```
Region(num_blocks=1)
```

So we might need to talk to the xDSL people about how this works.

Reply:
It would be great to add these if you have the motivation